### PR TITLE
Convert 4.7 sample to use MSB

### PR DESCRIPTION
**Pull request by richlander**
* This is the first pass on creating MSB samples.
* Let's validate this one and then I'll update the other ones.
* We also need to decide what we're going to do w/rt RS1 and RS3 as base images.

### CODE REVIEW

**Comment by MichaelSimons** [comment]
Regarding `We also need to decide what we're going to do w/rt RS1 and RS3 as base images.`, I think we should be using the manifest backed tags just as you done.  This will allow the sample to be run on either an RS1 or RS3 host.

File: dotnetapp-4.7/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,10 @@
+FROM microsoft/dotnet-framework-build:4.7.1 AS build-env
```
Comment (MichaelSimons) [inline dotnetapp-4.7/Dockerfile]: I'm thinking a `.dockerignore` should be created next to the Dockerfile to ignore any `bin` and `obj` folders in the case where the project may have been built locally.

**Comment by richlander** [comment]
> manifest-backed tags

I wrote something ambiguous. I mean my statement on RS1 vs. RS3 w/rt the sample image we actually push to DH. For the actual source file on GH, I agree with you.

> .dockerignore

Yes

How about the restore step? I did it with msbuild. Is that the best?

**Comment by MichaelSimons** [comment]
> I wrote something ambiguous. I mean my statement on RS1 vs. RS3 w/rt the sample image we actually push to DH. For the actual source file on GH, I agree with you.

I thought we would do the same thing we do for the [microsoft/dotnet-samples](https://hub.docker.com/r/microsoft/dotnet-samples/).  We build with the [sample Dockerfile](https://github.com/dotnet/dotnet-docker-samples/blob/master/dotnetapp-prod/Dockerfile) once per platform - RS1 and RS3 and then surface the resulting images via a manifest tag.  The build definition will handle the multiple builds.  Therefore what you have proposed will work just fine.

> How about the restore step? I did it with msbuild. Is that the best?

Given this is an SDK based project, is there any reason to not use the .NET Core SDK?  That is what the readme currently suggests - https://github.com/Microsoft/dotnet-framework-docker-samples/tree/master/dotnetapp-4.7#script

Were you going to update the readme as well?  With the introduction of multi-stage build, there is no longer the need to install VS and build locally.

**Pull request closed without merging**